truffle/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/parser/NodeParser.java

```diff
@@ -463,6 +463,11 @@ private NodeData parseNodeData(TypeElement templateType, List<TypeElement> typeH
                 } else if (names.contains(name)) {
                     field.addError(ElementUtils.getAnnotationValue(mirror, "name"), "Duplicate field name '%s'.", name);
                 }
+
+                if (type == null) {
+                    throw new AssertionError("The type specified in the NodeField annotation cannot be found. A possible cause could be a missing import statement for the type.");
```

Comment:
I think that there should be a custom exception. Truffle (and the DSL) should define a hierarchy of custom exceptions. 

BTW: the same applies to https://github.com/graalvm/truffle/blob/master/truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/frame/FrameDescriptor.java#L134